lib/vector-core/src/event/metric.rs

```diff
@@ -488,10 +501,8 @@ impl MetricValue {
     /// Add another same value to this.
     pub fn add(&mut self, other: &Self) {
         match (self, other) {
-            (Self::Counter { ref mut value }, Self::Counter { value: value2 }) => {
-                *value += value2;
-            }
-            (Self::Gauge { ref mut value }, Self::Gauge { value: value2 }) => {
+            (Self::Counter { ref mut value }, Self::Counter { value: value2 })
+            | (Self::Gauge { ref mut value }, Self::Gauge { value: value2 }) => {
```

Comment:
TIL I didn't expect we could merge cases where the values pointed inside different structs.

Reply:
I didn't realize it could be done either until clippy dinged us on it.